pkgs/applications/misc/polar-bookshelf/default.nix

```diff
@@ -0,0 +1,99 @@
+{ stdenv, lib, makeWrapper, fetchurl
+, dpkg, wrapGAppsHook
+, gtk3, cairo, gnome2, atk, gdk_pixbuf, glib
+, at-spi2-atk, dbus, libX11, libxcb, libXi
+, libXcursor, libXdamage, libXrandr, libXcomposite
+, libXext, libXfixes, libXrender, libXtst, libXScrnSaver
+, nss, nspr, alsaLib, cups, fontconfig, expat
+, libudev0-shim, glibc, curl, openssl, nghttp2, gnome3 }:
+
+
+stdenv.mkDerivation rec {
+  name = "polar-bookshelf-${version}";
+  version = "1.0.2";
+
+  # fetching a .deb because there's no easy way to package this Electron app
+  src = fetchurl {
+    url = "https://github.com/burtonator/polar-bookshelf/releases/download/v${version}/polar-bookshelf-${version}-amd64.deb";
+    sha256 = "0fz34qvd7zp5rd5rqljwhq8ya9x29cq5h8axb82z369nkbi1xpbs";
+  };
+
+  buildInputs = [
+    gnome3.gsettings_desktop_schemas
+    glib
+    gtk3
+  ];
+
+  nativeBuildInputs = [ 
+    wrapGAppsHook
+    makeWrapper 
+    dpkg
+  ];
+
+  libPath = lib.makeLibraryPath [
+      gtk3
+      cairo
+      gnome2.pango
+      atk
+      gdk_pixbuf
+      glib
+      at-spi2-atk
+      dbus
+      libX11
+      libxcb
+      libXi
+      libXcursor
+      libXdamage
+      libXrandr
+      libXcomposite
+      libXext
+      libXfixes
+      libXrender
+      libXtst
+      libXScrnSaver
+      nss
+      nspr
+      alsaLib
+      cups
+      fontconfig
+      expat
+  ];
+
+  runtimeLibs = lib.makeLibraryPath [ libudev0-shim glibc curl openssl nghttp2 ];
+
+  unpackPhase = "dpkg-deb -x $src .";
+
+  installPhase = ''
+    mkdir -p $out/share/polar-bookshelf
+    mkdir -p $out/bin
+    mkdir -p $out/lib
+
+    mv opt/Polar\ Bookshelf/* $out/share/polar-bookshelf
+    mv $out/share/polar-bookshelf/*.so $out/lib
+
+    mv usr/share/* $out/share/
+
+    ln -s $out/share/polar-bookshelf/polar-bookshelf $out/bin/polar-bookshelf
+  '';
+
+  preFixup = ''
+    for lib in $out/lib/*.so; do
+      patchelf --set-rpath "$out/lib:${libPath}" $lib
+    done
+
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+             --set-rpath "$out/lib:${libPath}" \
+             $out/bin/polar-bookshelf 
```

Comment:
Can you check if you can use `autoPatchelfHook` here? It is an automatic variant of patchelf and is a bit easier to maintain and has better checks if something is missing.

Reply:
pkgs/applications/misc/masterpdfeditor/default.nix has an example.